### PR TITLE
Update jenkins cicd documentation

### PR DESCRIPTION
**Pull request by pohek321**

### CODE REVIEW

**Comment by pohek321** [comment]
@sunkickr can you let me know why these checks are failing? Not quite sure I understand what's going on there.

File: astro/ci-cd.md [hunk 1]
```diff
@@ -279,41 +279,107 @@ To complete this setup, you need:
...
+
+To automate code deploys to multiple Deployments using [Jenkins](https://www.jenkins.io/), complete the following setup in a Git-based repository hosting an Astro project:
+
+1. In Jenkins, add the following environment variables:
+
+   :::info
+
+   To set environment variables in Jenkins, from the Jenkins Dashboard navigate to `Manage Jenkins >> Configure System >> Global Properties >> Environment Variables >> Add`
+
+   :::
```
Comment (jwitz) [inline astro/ci-cd.md:329]:
```suggestion
```

File: astro/ci-cd.md [hunk 2]
```diff
@@ -279,41 +279,107 @@ To complete this setup, you need:
...
+
+    - `PROD_ASTRONOMER_KEY_ID`: Your Production Deployment API key ID
+    - `PROD_ASTRONOMER_KEY_SECRET`: Your Production Deployment API key secret
+    - `PROD_DEPLOYMENT_ID`: The Deployment ID of your Production Deployment
+    - `DEV_ASTRONOMER_KEY_ID`: Your Development Deployment API key ID
+    - `DEV_ASTRONOMER_KEY_SECRET`: Your Development Deployment API key secret
+    - `DEV_DEPLOYMENT_ID`: The Deployment ID of your Development Deployment
+
+    Be sure to set the values for your API credentials as secret.
```
Comment (jwitz) [inline astro/ci-cd.md:338]:
```suggestion

    To set environment variables in Jenkins, on the Jenkins Dashboard go to **Manage Jenkins** > **Configure System** > **Global Properties** > **Environment Variables** > **Add**.

    Be sure to set the values for your API credentials as secret.
```

File: astro/ci-cd.md [hunk 3]
```diff
@@ -279,41 +279,107 @@ To complete this setup, you need:
...
+                          echo "This git branch ${env.GIT_BRANCH} is not configured in this pipeline."
+                      }
+                  }
+              }
+           }
+           stage('Deploy to Astronomer') {
+             steps {
+               script {
+                 sh 'curl -LJO https://github.com/astronomer/astro-cli/releases/download/v1.1.0/astro_1.1.0_linux_amd64.tar.gz'
+                 sh 'tar xzf astro_1.1.0_linux_amd64.tar.gz'
```
Comment (jwitz) [inline astro/ci-cd.md:368]:
```suggestion
                 sh 'curl -LJO https://github.com/astronomer/astro-cli/releases/download/v${siteVariables.cliVersion}/astro_${siteVariables.cliVersion}_linux_amd64.tar.gz'
                 sh 'tar xzf astro_${siteVariables.cliVersion}_linux_amd64.tar.gz'
```

File: astro/ci-cd.md [hunk 4]
```diff
@@ -279,41 +279,107 @@ To complete this setup, you need:
...
+                 sh "./astro deploy ${DEPLOYMENT_ID} -f"
```
Comment (jwitz) [inline astro/ci-cd.md:369]:
```suggestion
                 sh "./astro deploy ${siteVariables.deploymentid} -f"
```

File: astro/ci-cd.md [hunk 5]
```diff
@@ -279,41 +279,107 @@ To complete this setup, you need:
...
+           cleanWs()
+         }
        }
       }
-    }
-    `}</code></pre>
+   }`</code></pre>
 
-    This Jenkinsfile triggers a code push to Astro every time a commit or pull request is merged to the `main` branch of your repository.
+    This Jenkinsfile triggers a code push to your Astro deployments every time a commit or pull request is merged to the `dev` & `main` branch of your repository.
```
Comment (jwitz) [inline astro/ci-cd.md:382]:
```suggestion
    This Jenkinsfile triggers a code push to an Astro Deployment every time a commit or pull request is merged to the `dev` or `main` branch of your repository.
```

File: astro/ci-cd.md [hunk 6]
```diff
@@ -279,41 +279,104 @@ To complete this setup, you need:
...
+    - `DEV_ASTRONOMER_KEY_SECRET`: Your Development Deployment API key secret
+    - `DEV_DEPLOYMENT_ID`: The Deployment ID of your Development Deployment
+
+    To set environment variables in Jenkins, on the Jenkins Dashboard go to **Manage Jenkins** > **Configure System** > **Global Properties** > **Environment Variables** > **Add**.
+
+    Be sure to set the values for your API credentials as secret.
+
+2. At the root of your Git repository, add a [Jenkinsfile](https://www.jenkins.io/doc/book/pipeline/jenkinsfile/) that includes the following script:
+
+    <pre><code parentName="pre">`pipeline {
```
Comment (jwitz) [inline astro/ci-cd.md:339]:
```suggestion
    <pre><code parentName="pre">{`pipeline {
```

File: astro/ci-cd.md [hunk 7]
```diff
@@ -279,41 +279,104 @@ To complete this setup, you need:
...
-           when {
-            expression {
-              return env.GIT_BRANCH == "origin/main"
-            }
-           }
-           steps {
-             script {
-                   sh 'curl -sSL install.astronomer.io | sudo bash -s'
-                   sh 'astro deploy -f'
+    <pre><code parentName="pre">`pipeline {
```
Comment (jwitz) [inline astro/ci-cd.md:292]:
```suggestion
    <pre><code parentName="pre">{`pipeline {
```

File: astro/ci-cd.md [hunk 8]
```diff
@@ -279,41 +279,104 @@ To complete this setup, you need:
...
          }
+       post {
+         always {
+           cleanWs()
+         }
        }
-     post {
-       always {
-         cleanWs()
+   }`</code></pre>
```
Comment (jwitz) [inline astro/ci-cd.md:315]:
```suggestion
   }`}</code></pre>
```

File: astro/ci-cd.md [hunk 9]
```diff
@@ -279,41 +279,104 @@ To complete this setup, you need:
...
+         }
+       post {
+         always {
+           cleanWs()
+         }
        }
       }
-    }
-    `}</code></pre>
+   }`</code></pre>
```
Comment (jwitz) [inline astro/ci-cd.md:377]:
```suggestion
   }`}</code></pre>
```

**Pull request merged**